test/tests.py

```diff
@@ -990,6 +993,46 @@ def test(self):
                         dmlc_extraargs = ['--dep', 'T_dmldep.dmldep'],
                         api_version=latest_api_version))
 
+class DmlDepNinja(DMLFileTestCase):
+    '''Stability test, '''
+    __slots__ = ()
+    def __init__(self, path, filename, **info):
+        DMLFileTestCase.__init__(
+            self, "/".join(path), filename, **info)
+    def test(self):
+        if not isdir(self.scratchdir):
+            os.makedirs(self.scratchdir)
+        flags = self.test_flags()
+        # Run dmlc
+        status = self.runlog("dmlc", self.run_dmlc, self.filename,
+                             self.dmlc_extraargs + flags.dmlc_flags)
+
+        if status != self.status:
+            self.print_logs('dmlc', self.dmlc_stdout, self.dmlc_stderr)
+            raise TestFail("dmlc status=%d (expected %d)"
+                           % (status, self.status))
+        # test .dmldep
+        with open(self.cfilename + '.dmldep') as f:
+            target_prereqs = self.parse_depfile(f)
+        c_target = 'T_%s.c' % (self.shortname,)
+        dmldep_target = 'T_%s.dmldep' % (self.shortname,)
+        non_lib = {os.path.normpath(p) for p in target_prereqs[c_target]
+                   if not p.startswith(project_host_path())}
+        # Main .c target depends on main DML file and all DML imports.
+        # We test that all non-library DML files are included, as well as
+        # one library file
+        self.expect_equal_sets(non_lib, {
+            os.path.normpath(join(testdir, '1.2', 'misc', f + '.dml'))
+            for f in ['T_import_rel_1', 'rel/a/x', 'rel/a/y',
+                      'rel/z', 'rel/misc/rel/z']})
+
+        assert list(target_prereqs.keys()) == [c_target], "Ninja deps should have only .c target"
```

Comment:
I added a test here. There is a minimal test for CMake build `test/t1019_project_cmake`. However, we are missing some critical pieces to add more. In particular `project-setup` arguments to generate devices need to understand `--cmake` option to create `CMakeLists.txt` file instead of `Makefile`. Once this is done, we can add integrations tests for CMake build to cover:
* Ninja and Unix Makefiles generators
* Build for DML, C, and other devices

Reply:
@andreashedstrom-intel any comments on roadmap for t1019 or `project-setup --cmake`?

Reply:
I don't understand what @adklimki is referring to here; the t1019 test *does* use --cmake to create the project with top-level CMakeLists.txt for building with CMake instead of GNU Make.

The project-setup --copy-module can be used to copy modules into the project, but as t1019 currently builds *all* converted modules distributed by Simics Base there is (currently) no need to add this. If we add new dummy devices that we don't want to distribute with Simics Base these devices can be manually copied/symlinked to the project directly from the outer repo.

t1019 currently assumes we run with -G Ninja but can easily be updated to run with -G Unix Makefiles (I'll add this as a separate PR; we should probably have this to explicitly cover the case where we want to support both generators). 

As we build all converted modules, we already build DML, C, C++ and python based modules in t1019.

Reply:
Since we don't have a specification of Ninja's depfile requirements, it would be great with some kind of complementary smoke test that DML build deps work.

Reply:
@andreashedstrom-intel I mean this command creates a Makefile, not CMakeLists.txt:
```
> ./bin/project-setup --cmake --dml-device XXX
Skeleton for module 'XXX' successfully created.
> ls -la modules/XXX
total 24
drwxr-x--- 3 adklimki users 4096 Jan  4 12:08 ./
drwxr-x--- 3 adklimki users 4096 Jan  4 12:08 ../
-rw-r----- 1 adklimki users  402 Jan  4 12:08 Makefile
-rw-r----- 1 adklimki users 1724 Jan  4 12:08 XXX.dml
-rw-r----- 1 adklimki users  468 Jan  4 12:08 module_load.py
drwxr-x--- 2 adklimki users 4096 Jan  4 12:08 test/
```

I see there are tests that use these templates to test the build scripts.

Reply:
ah, ok, yes now *that* is indeed missing. Current flow just supports building already converted modules. And converting a skeleton device is straight forward and easily done and should work out-of-the-box by invoking cmake/gmake-to-cmake.py on the modules/XXX/Makefile to produce the modules/XXX/CMakeLists.txt.

But yes, I agree, this should be done by project-setup to save the user from this. Not sure if we need a special --cmake flag here. We could simply generate both CMakeLists.txt and Makefile and let the user choose which one to keep. What do you think @mandolaerik ?

The t1019 now builds both with Ninja and Unix Makefiles and as mentioned it builds all converted modules so it's both C, C++, DML and Python. It does not run any tests though. We probably want a dedicated suite for smoking out problems with the DML deps as Erik suggested. We already know C/C++ deps work, as that works out-of-the-box in CMake.

Reply:
It should be easy to write a smoke test that either writes a hardcoded CMakeLists.txt, or runs a conversion script to produce one. I assume cmake is still in tech preview so we should not yet create it by default from project-setup. We should eventually add this, and then we can remove the explicit generation from the smoke test.